### ISSUE TITLE
AutoSpawner should check for clearance when building Iron Golems and Withers

### ISSUE THREAD

**Issue by omninope** [kami-blue/client#1605]
**Is your feature request related to a problem? Please describe.**
This is a problem which limits the usability of the module.
AutoSpawner builds invalid structures for Iron Golems and Withers when used on uneven terrain, and that includes basically all Minecraft terrain.

**Describe the solution you'd like**
AutoSpawner should check for clearance under and above the "arms" of Iron Golem structures and under the "arms" of Wither structures when building because Minecraft requires this blocks to be air blocks. Currently this module builds the structures even if they are invalid.

As demonstrated in the images below (taken from Minecraft Wiki) the blocks in white should be air for the structure to be valid. Also liquids, grass, flowers, mushrooms and similar makes the structure invalid.

![Screenshot_20201118_231023](https://user-images.githubusercontent.com/69328977/99612319-42e24a80-2a0d-11eb-8491-5643d7975431.png)
![Screenshot_20201118_231028](https://user-images.githubusercontent.com/69328977/99612330-47a6fe80-2a0d-11eb-9ad1-bfbca3a2e166.png)

Basically it should check if there is a 3x3x1 space of air.

**Describe alternatives you've considered**
Only using the module on perfetly flat and even terrain.

**Additional context**
No.
